exercises/diffie-hellman/canonical-data.json

```diff
@@ -0,0 +1,67 @@
+{
+  "exercise": "diffie-hellman",
+  "version": "1.0.0",
+  "comments": [
+    "Optional tests to consider:",
+    "* Validation of parameters:",
+    "  Validate that `p`, `g` are valid.",
+    "  Validate that keys given as inputs are valid.",
+    "  Resources that show what happens if parameters are not validated:",
+    "  http://cryptopals.com/sets/5/challenges/34",
+    "  http://cryptopals.com/sets/5/challenges/35",
+    "* Large numbers:",
+    "  Although the calculations fundamentally do not require large numbers,",
+    "  this is a reasonable real-world use for them",
+    "  and it may be instructive to have an exercise on their use.",
+    "  Consult tracks with this exercise (such as the Go track) for possible inputs to use."
+  ],
+  "cases": [
+    {
+      "description": "private key is in range 1 .. p",
+      "property": "privateKeyIsInRange",
+      "input": {},
+      "expected": {}
+    },
+    {
+      "description": "private key is random",
+      "property": "privateKeyIsRandom",
+      "input": {},
+      "expected": {}
```

Comment:
I'm definitely not here to tell individual tracks how they test this.

There is no precedent. Of the exercises that test randomness:

* `robot-name` also has no JSON file yet.
* `simple-cipher`'s https://github.com/exercism/problem-specifications/pull/966 received no suggestions.
* `diffie-hellman`'s https://github.com/exercism/problem-specifications/pull/944#discussion_r147567932 received no suggestions.

Therefore, I decided what it should look like.

Reply:
I don't think there is any good way to model an expected random outcome.

Reply:
As noted above, there is currently no precedent for creating canonical data to test randomness. As such, this exercise will become that precedent, so we will want to think carefully about the best way to do this that is also reusable in other instances (i.e. `robot-name` and company)

Reply:
I'll *start* with `"expected": {"random": true}`, but I still get the feeling it could be better. I will write a comment later saying why.